src/threading.c

```diff
@@ -237,11 +243,18 @@ void ti_threadfun(void *arg)
 #endif
 
         if (work) {
-            if (work->command == TI_THREADWORK_DONE)
+            if (work->command == TI_THREADWORK_DONE) {
                 break;
-            else if (work->command == TI_THREADWORK_RUN)
+            }
+            else if (work->command == TI_THREADWORK_RUN) {
                 // TODO: return value? reduction?
+                // TODO: before we support getting return value from
+                //       the work, we don't need to enter GC managed region
+                //       when starting the work.
+                int8_t state = jl_gc_managed_enter();
```

Comment:
should this `state` be added to the exception state (jl_enter / jl_leave / jl_rethrow)?